### PR TITLE
PRIME-2400 Update Practitioner Verification logic

### PR DESCRIPTION
**Pull request by bergomi02**
Test Link:
https://pr-2284.pharmanetenrolment.gov.bc.ca/info

Note: 
- Admin Event Log has been updated to show the effective date
- In order to test auto-adjudication, you need to disable couple validation rule around the address because the Dev environment does not supply the address data. 
image
- To test the situation where enrollee has two licence records (e.g. RX and R9) locally, you can update DummyCollegeLicenceClient.GetCollegeRecordAsync() to return the specific licence records. I have tested it with BCSC Test 40 with the code below:
image
The system should store the prefix of the licence with latest effective date.

### CODE REVIEW

**Comment by neophyte57** [comment]
I experienced strange behavior with this enrollment:  For some reason I don't see Event Logs showing calls to PharmaNet API

image

I deleted enrollment and tried again with same result  🤔

File: prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs [hunk 1]
```diff
@@ -108,30 +109,45 @@ record = await _collegeLicenceClient.GetCollegeRecordAsync(cert.Prefix, cert.Lic
                 if (record == null)
                 {
                     await _businessEventService.CreatePharmanetApiCallEventAsync(enrollee.Id, cert.Prefix, cert.LicenseNumber, "Record not found in Pharmanet.");
-                    if (cert.NonPrescribingPrefix != null)
+                }
+                else
+                {
+                    await _businessEventService.CreatePharmanetApiCallEventAsync(enrollee.Id, cert.Prefix, cert.LicenseNumber,
+                        $"A record was found in Pharmanet with effective date {record.EffectiveDate:dd MMM yyy}.");
```
Comment (neophyte57) [inline prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs:116]: "Pharmanet" should have proper casing "PharmaNet" (https://www2.gov.bc.ca/gov/content/health/health-drug-coverage/pharmacare-for-bc-residents/pharmanet)

**Comment by bergomi02** [comment]
> For this enrollment, where both records are Non-Practicing
> 
> image
> 
> why is the older "R9" record considered higher priority?
> 
> image

R9 is the default prefix. Checked the database and the prefix is not set which is the expected behavior. 
image

File: prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs [hunk 2]
```diff
@@ -99,39 +100,70 @@ record = await _collegeLicenceClient.GetCollegeRecordAsync(cert.Prefix, cert.Lic
...
                 if (record == null)
                 {
-                    await _businessEventService.CreatePharmanetApiCallEventAsync(enrollee.Id, cert.Prefix, cert.LicenseNumber, "Record not found in Pharmanet.");
-                    if (cert.NonPrescribingPrefix != null)
+                    await _businessEventService.CreatePharmanetApiCallEventAsync(enrollee.Id, cert.Prefix, cert.LicenseNumber, "Record not found in PharmaNet.");
+                }
+                else
+                {
+                    await _businessEventService.CreatePharmanetApiCallEventAsync(enrollee.Id, cert.Prefix, cert.LicenseNumber,
+                        $"A record was found in PharmaNet with effective date {record.EffectiveDate:dd MMM yyy} and status {record.Status}.");
```
Comment (neophyte57) [inline prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs:116]: Minor thing, can the date format be `d MMM yyyy` (two occurrences) 

image

File: prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs [hunk 3]
```diff
@@ -99,39 +100,70 @@ record = await _collegeLicenceClient.GetCollegeRecordAsync(cert.Prefix, cert.Lic
...
                             }
                         }
-                        catch (PharmanetCollegeApiException)
+                        else
                         {
-                            enrollee.AddReasonToCurrentStatus(StatusReasonType.PharmanetError, $"{cert.NonPrescribingPrefix}-{cert.LicenseNumber}");
-                            await _businessEventService.CreatePharmanetApiCallEventAsync(enrollee.Id, cert.NonPrescribingPrefix, cert.LicenseNumber, "An error occurred calling the Pharmanet API.");
-                            passed = false;
-                            continue;
+                            await _businessEventService.CreatePharmanetApiCallEventAsync(enrollee.Id, cert.NonPrescribingPrefix, cert.LicenseNumber, "Record not found in Pharmanet.");
```
Comment (neophyte57) [inline prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs:157]: `PharmaNet` (casing)

File: prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs [hunk 4]
```diff
@@ -99,39 +100,70 @@ record = await _collegeLicenceClient.GetCollegeRecordAsync(cert.Prefix, cert.Lic
...
                         }
                     }
+                    catch (PharmanetCollegeApiException)
+                    {
+                        enrollee.AddReasonToCurrentStatus(StatusReasonType.PharmanetError, $"{cert.NonPrescribingPrefix}-{cert.LicenseNumber}");
+                        await _businessEventService.CreatePharmanetApiCallEventAsync(enrollee.Id, cert.NonPrescribingPrefix, cert.LicenseNumber, "An error occurred calling the Pharmanet API.");
```
Comment (neophyte57) [inline prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs:163]: `PharmaNet` (casing)

File: prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs (hunk 2 above)
Comment (bergomi02) [reply prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs]: fixed.

File: prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs (hunk 3 above)
Comment (bergomi02) [reply prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs]: fixed.

File: prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs (hunk 4 above)
Comment (bergomi02) [reply prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs]: fixed.

**Review by neophyte57 — APPROVED** [review]
Long road but nice work!

**Pull request merged**